apps/heft/src/schemas/heft-legacy.schema.json

```diff
@@ -0,0 +1,214 @@
+                          "items": {
+                            "type": "string",
+                            "pattern": "[^\\\\]"
+                          }
+                        },
+
+                        "flatten": {
+                          "type": "boolean",
+                          "description": "Normally, when files are selected under a child folder, a corresponding folder will be created in the destination folder.  Specify flatten=true to discard the source path and copy all matching files to the same folder.  If two files have the same name an error will be reported. The default value is false."
+                        },
+
+                        "hardlink": {
+                          "type": "boolean",
+                          "description": "If true, filesystem hard links will be created instead of copying the file.  Depending on the operating system, this may be faster. (But note that it may cause unexpected behavior if a tool modifies the link.)  The default value is false."
+                        }
+                      }
+                    }
+                  }
+                }
+              },
+              {
+                "required": ["scriptPath"],
+                "properties": {
+                  "actionKind": {
+                    "type": "string",
+                    "enum": ["runScript"]
+                  },
+
+                  "heftEvent": {
+                    "type": "string",
+                    "enum": ["pre-compile", "compile", "bundle", "post-build", "test"]
+                  },
+
+                  "scriptPath": {
+                    "type": "string",
+                    "description": "Path to the script that will be run, relative to the project root.",
+                    "items": {
+                      "type": "string",
+                      "pattern": "[^\\\\]"
+                    }
+                  },
+
+                  "scriptOptions": {
+                    "type": "object",
+                    "description": "Optional parameters that will be passed to the script at runtime.",
+                    "patternProperties": {
+                      "^.*$": { "$ref": "#/definitions/anything" }
+                    }
+                  }
+                }
+              }
+            ]
+          }
+        ]
+      }
+    },
+
+    "heftPlugins": {
+      "type": "array",
+      "description": "Defines heft plugins that are used by a project.",
+
+      "items": {
+        "type": "object",
+        "required": ["plugin"],
+        "properties": {
+          "plugin": {
+            "description": "Path to the plugin package, relative to the project root.",
+            "type": "string",
+            "pattern": "[^\\\\]"
+          },
+
+          "options": {
+            "type": "object"
+          }
+        }
+      }
+    }
```

Comment:
Do we actually care that the stuff inside `"heftPlugins"` and `"eventActions"` matches the old schema, or just if it kinda looks like an old config file?

Reply:
`heftPlugins` actually exists in the new format, even though the contents of it has changed. That's why I'm validating the entire config instead of just looking for specific properties in the raw file